pkg/labels/labels.go

```diff
@@ -383,20 +383,18 @@ func (l *Label) HasKey(target *Label) bool {
 		tc := target.cidr
 		if tc == nil {
 			v, err := LabelToPrefix(target.Key)
-			if err != nil {
+			if err == nil {
```

Comment:
wow, we're damn lucky this never caught us. Lucky that we always parsed the labels earlier. Whew.